src/PublicApi/WebFileSystem.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.eShopWeb.PublicApi.CatalogItemEndpoints;
+using Newtonsoft.Json;
+
+namespace Microsoft.eShopWeb.PublicApi
+{
+    public class WebFileSystem: IFileSystem
+    {
+        private readonly HttpClient _httpClient;
+        private readonly string _url;
+        public const string AUTH_KEY = "AuthKeyOfDoomThatMustBeAMinimumNumberOfBytes";
+
+        public WebFileSystem(string url)
+        {
+            _httpClient = new HttpClient();
+            _httpClient.DefaultRequestHeaders.Add("auth-key", AUTH_KEY);
+            _url = url;
+        }
+
+        public async Task<bool> UploadFile(string fileName, byte[] fileData)
+        {
+            if (!fileData.IsValidImage(fileName))
+            {
+                return false;
+            }
+
+            return await UploadToWeb(fileName, fileData);
+        }
+
+        public static async Task<bool> SavePicture(string pictureName, string pictureBase64)
```

Comment:
Why is this static? You can't use it with DI or the IFileSystem interface.

Reply:
Done